neptun/src/device/api.rs

```diff
@@ -148,26 +114,31 @@ pub fn api_exec<R: Read, W: Write>(
     reader: &mut BufReader<R>,
     writer: &mut BufWriter<W>,
 ) {
+    let mut status = 0;
     let mut cmd = String::new();
-    if reader.read_line(&mut cmd).is_ok() {
-        cmd.pop(); // pop the new line character
-        let status = match d.closed {
+    while status == 0 && reader.read_line(&mut cmd).is_ok_and(|n| n > 0) {
+        status = match d.closed {
             true => ENOENT,
             false => match cmd.as_ref() {
                 // Only two commands are legal according to the protocol, get=1 and set=1.
-                "get=1" => api_get(writer, d),
-                "set=1" => api_set(reader, d),
+                "get=1\n" => api_get(reader, writer, d),
```

Comment:
Do we really want to deal with all these new lines manually? What do you think about using something like `cmd.trim()` to sanitize the input, and `cmd.split_whitespace()` if there are multiple commands?

Reply:
Yes, we do. It's not about striping them. Stripping was (incorrectly) done earlier via `pop`. Here the point is that the \n is required. So sending "get=1[EOF]" is considered an incorrect command. It's just to keep compatibility with wireguard-go, nothing more.